portal-impl/src/com/liferay/portal/zip/SevenZipWriterImpl.java

```diff
@@ -0,0 +1,230 @@
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.portal.zip;
+
+import com.liferay.exportimport.kernel.lar.ExportImportThreadLocal;
+import com.liferay.petra.io.StreamUtil;
+import com.liferay.petra.memory.DeleteFileFinalizeAction;
+import com.liferay.petra.memory.FinalizeManager;
+import com.liferay.petra.string.StringBundler;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.util.FileUtil;
+import com.liferay.portal.kernel.util.SystemProperties;
+import com.liferay.portal.kernel.uuid.PortalUUIDUtil;
+import com.liferay.portal.kernel.zip.ZipWriter;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+
+import java.nio.file.FileSystem;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.StandardCopyOption;
+import java.nio.file.StandardOpenOption;
+import java.nio.file.attribute.BasicFileAttributes;
+
+import org.apache.commons.compress.archivers.sevenz.SevenZArchiveEntry;
+import org.apache.commons.compress.archivers.sevenz.SevenZOutputFile;
+
+/**
+ * @author Hong Vo
+ */
+public class SevenZipWriterImpl implements ZipWriter {
+
+	public SevenZipWriterImpl() {
+		this(
+			new File(
+				StringBundler.concat(
+					SystemProperties.get(SystemProperties.TMP_DIR),
+					StringPool.SLASH, PortalUUIDUtil.generate())));
+
+		FinalizeManager.register(
+			_directory,
+			new DeleteFileFinalizeAction(_directory.getAbsolutePath()),
+			FinalizeManager.PHANTOM_REFERENCE_FACTORY);
+	}
+
+	public SevenZipWriterImpl(File directory) {
+		_directory = directory.getAbsoluteFile();
+
+		_path = _directory.toPath();
+
+		_file = null;
+	}
+
+	@Override
+	public void addEntry(String name, byte[] bytes) throws IOException {
+		if (bytes == null) {
+			return;
+		}
+
+		FileSystem fileSystem = _path.getFileSystem();
+
+		Path path = fileSystem.getPath(
+			StringBundler.concat(_path.toString(), StringPool.SLASH, name));
+
+		Path parentPath = path.getParent();
+
+		if (parentPath != null) {
+			Files.createDirectories(parentPath);
+		}
+
+		Files.write(
+			path, bytes, StandardOpenOption.CREATE,
+			StandardOpenOption.TRUNCATE_EXISTING, StandardOpenOption.WRITE);
+	}
+
+	@Override
+	public void addEntry(String name, InputStream inputStream)
+		throws IOException {
+
+		if (inputStream == null) {
+			return;
+		}
+
+		if (ExportImportThreadLocal.isExportInProcess()) {
+			addEntry(name, StreamUtil.toByteArray(inputStream));
+
+			return;
+		}
+
+		FileSystem fileSystem = _path.getFileSystem();
+
+		Path path = fileSystem.getPath(
```

Comment:
This is the path of the file entry? `path` and `_path` make me feel confused. What do you think about changing it?